intelmq/bots/collectors/shadowserver/collector_reports_api.py

```diff
@@ -138,15 +138,18 @@ def process(self):
                 self.logger.debug('Processed file %r (fixed: %r) already.', filename, filename_fixed)
                 continue
             self.logger.debug('Processing file %r (fixed: %r).', filename, filename_fixed)
-            reportdata = self._report_download(item['id'])
-            report = self.new_report()
-            report.add('extra.file_name', filename_fixed)
-            report.add('raw', reportdata)
-            self.send_message(report)
-            self.cache.set(filename, 1)
-            self.logger.debug('Sent report: %r (fixed: %r, size: %.3g KiB).', filename, filename_fixed,
+            try:
+                reportdata = self._report_download(item['id'])
+                report = self.new_report()
+                report.add('extra.file_name', filename_fixed)
+                report.add('raw', reportdata)
+                self.send_message(report)
+                self.cache.set(filename, 1)
+                self.logger.debug('Sent report: %r (fixed: %r, size: %.3g KiB).', filename, filename_fixed,
                               len(reportdata) / 1024)  # TODO: Replace by a generic size-conversion function
-            reports_downloaded += 1
+                reports_downloaded += 1
+            except:
```

Comment:
That's a bare except which can hide unrelated errors. Does using `requests.exceptions.ReadTimeout` alone work as well?

Reply:
If any error is received then the whole bot stops. It is possible to use ReadTimeout, but if we get another error it will also stop the bot.

Reply:
We can add more exceptions here, but a bare except is a no-go. That hides completely unrelated errors, an extreme example is `MemoryError`.